Naming/SmartUniqueNamer.php

```diff
@@ -17,10 +17,25 @@ public function name($object, PropertyMapping $mapping): string
     {
         $file = $mapping->getFile($object);
         $originalName = $file->getClientOriginalName();
-        $originalExtension = \pathinfo($originalName, PATHINFO_EXTENSION);
-        $originalBasename = \basename($originalName, '.'.$originalExtension);
+        $originalExtension = \strtolower(\pathinfo($originalName, PATHINFO_EXTENSION));
+        $originalBasename = \strtolower(\basename($originalName, '.'.$originalExtension));
         $originalBasename = Transliterator::transliterate($originalBasename);
+        $uniqId = \str_replace('.', '', \uniqid('-', true));
+        $uniqExtension = \sprintf('%s.%s', $uniqId, $originalExtension);
+        $smartName = \sprintf('%s%s', $originalBasename, $uniqExtension);
+        // Check if smartName is an acceptable size (some filesystems accept a max of 255)
+        if (\strlen($smartName) <= 255) {
+            return $smartName;
+        }
 
-        return \sprintf('%s%s.%s', $originalBasename, \str_replace('.', '', \uniqid('-', true)), $originalExtension);
+        // Shorten the basename to fit into 255 (excluding the unique extension)
+        $diffSize = (255 - \strlen($uniqExtension)) - \strlen($originalBasename);
```

Comment:
I understand that we don't want to cut out the extension, but I think that diff size should be calculated by taking extension itself into account

Reply:
Fair enough, I assume there is no issue then using the `substr` function on the extension then to remove the excess?

Reply:
OK, let's take a little example: suppose that $smartName is 256 chars long, and extension is typical 3 letters + dot (like ".pdf"). We can deduct than $diffSize will be 252 (= 256 - 3 - 1) chars and so will fail check against 255.

Reply:
This should be taken care of in commit [a26773c](https://github.com/aewens/VichUploaderBundle/commit/a26773ce1bf2533195b862fec9951b93b375d638#diff-43549d537b7d4f88517af8b366fd4a62).